applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h

```diff
@@ -166,67 +166,72 @@ class ALMFrictionalMortarConvergenceCriteria
         IndexType is_converged_active = 0;
         IndexType is_converged_slip = 0;
 
+        // We get the process info
         ProcessInfo& r_process_info = rModelPart.GetProcessInfo();
-        const double common_epsilon = r_process_info[INITIAL_PENALTY];
-        const double scale_factor = r_process_info[SCALE_FACTOR];
-        const double tangent_factor = r_process_info[TANGENT_FACTOR];
 
-        NodesArrayType& nodes_array = rModelPart.GetSubModelPart("Contact").Nodes();
-
-        #pragma omp parallel for reduction(+:is_converged_active, is_converged_slip)
-        for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i) {
-            auto it_node = nodes_array.begin() + i;
-
-            const double epsilon = it_node->Has(INITIAL_PENALTY) ? it_node->GetValue(INITIAL_PENALTY) : common_epsilon;
+        // We check the active/inactive set during the first non-linear iteration or for the general semi-smooth case
+        if (rModelPart.Is(INTERACTION) || r_process_info[NL_ITERATION_NUMBER] == 1) {
```

Comment:
Using `INTERACTION` is misleading imo, why not use another flag? e.g. `SEMI_SMOOTH`?

Reply:
I am not using that flag and the number of flags is limited, I can not ask for a new one every time.

Reply:
ok
But you could create your own flags, [like I do](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MappingApplication/custom_utilities/mapper_flags.h) ...
Just saying

Reply:
That's a local flag, I cannot set a local flag on a model part (which contains the common flags, I think)